read.go

```diff
@@ -159,28 +159,58 @@ func (r *reader) readHeader() ([]*Element, error) {
 		return nil, err
 	}
 
+	metaElems := []*Element{maybeMetaLen} // TODO: maybe set capacity to a reasonable initial size
+	metaElementGroupLengthDefined := true
 	if maybeMetaLen.Tag != tag.FileMetaInformationGroupLength || maybeMetaLen.Value.ValueType() != Ints {
-		return nil, ErrorMetaElementGroupLength
+		if !r.opts.allowErrorMetaElementGroupLength {
+			return nil, ErrorMetaElementGroupLength
+		}
+		metaElementGroupLengthDefined = false
+		metaElems = append(metaElems, maybeMetaLen)
```

Comment:
does this add it twice? In L162, the maybeMetaLen item is already added to the metaElems slice.

Reply:
It does indeed.  Removed it.